pkg/injector/iptables.go

```diff
@@ -5,34 +5,50 @@ import (
 	"strconv"
 	"strings"
 
+	configv1alpha2 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
 	"github.com/openservicemesh/osm/pkg/constants"
 )
 
-// iptablesOutboundStaticRules is the list of iptables rules related to outbound traffic interception and redirection
-var iptablesOutboundStaticRules = []string{
-	// Redirects outbound TCP traffic hitting OSM_PROXY_OUT_REDIRECT chain to Envoy's outbound listener port
-	fmt.Sprintf("-A OSM_PROXY_OUT_REDIRECT -p tcp -j REDIRECT --to-port %d", constants.EnvoyOutboundListenerPort),
+func genIPTablesOutboundStaticRules(cfg configurator.Configurator) []string {
+	// iptablesOutboundStaticRules is the list of iptables rules related to outbound traffic interception and redirection
+	iptablesOutboundStaticRules := []string{
+		// Redirects outbound TCP traffic hitting OSM_PROXY_OUT_REDIRECT chain to Envoy's outbound listener port
+		fmt.Sprintf("-A OSM_PROXY_OUT_REDIRECT -p tcp -j REDIRECT --to-port %d", constants.EnvoyOutboundListenerPort),
+
+		// Traffic to the Proxy Admin port flows to the Proxy -- not redirected
+		fmt.Sprintf("-A OSM_PROXY_OUT_REDIRECT -p tcp --dport %d -j ACCEPT", constants.EnvoyAdminPort),
+	}
 
-	// Traffic to the Proxy Admin port flows to the Proxy -- not redirected
-	fmt.Sprintf("-A OSM_PROXY_OUT_REDIRECT -p tcp --dport %d -j ACCEPT", constants.EnvoyAdminPort),
+	localProxyMode := cfg.GetMeshConfig().Spec.Sidecar.LocalProxyMode
 
-	// For outbound TCP traffic jump from OUTPUT chain to OSM_PROXY_OUTBOUND chain
-	"-A OUTPUT -p tcp -j OSM_PROXY_OUTBOUND",
+	if localProxyMode == configv1alpha2.LocalProxyModePodIP {
+		// For envoy -> local service container proxying, send traffic to pod IP instead of localhost
+		iptablesOutboundStaticRules = append(iptablesOutboundStaticRules, fmt.Sprintf("-A OUTPUT -p tcp -o lo -d 127.0.0.1/32 -m owner --uid-owner %d -j DNAT --to-destination $POD_IP", constants.EnvoyUID))
+	}
```

Comment:
Can this not be added as a part of the dynamic rule set similar to other "dynamic" rules that are configurable?

Reply:
No because this rule must occur before line 33; we have to short-circuit envoy outbound traffic to localhost before it gets caught in the OSM_PROXY_OUTBOUND chain

Reply:
Oh I wasn't aware of that option! TIL - thanks!